### ISSUE TITLE
Feature: decouple index DB from specific RPC implementation

### ISSUE THREAD

**Issue by romanz** [romanz/electrs#554]
**Is your feature request related to a problem? Please describe.**
Today, electrs only supports JSONRPC, so adding more RPC implementations is tricky (e.g. https://github.com/romanz/electrs/issues/257).

**Describe the solution you'd like**
We should probably decouple all index build/query code into a separate library crate, allowing composing it with specific RPC protocols.

**Describe alternatives you've considered**
Using cargo features (similar to how BWT allows both Electrum RPC and [HTTP API](https://github.com/bwt-dev/bwt/#http-api)).

**Comment by Kixunil** [comment]
Yeah, I'd like to see moving closer towards [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) (also abstracting block/transaction retrieving and DB)
